airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/pull_request/pipeline.py

```diff
@@ -0,0 +1,396 @@
+            except UnknownObjectException:
+                # don't need to add it to the tree
+                logger.info(f"        {changed_file.path} not in parent: {base_sha}")
+                continue
+
+        # Update or new file addition or needed deletion
+        tree_elements.append(
+            InputGitTreeElement(
+                path=changed_file.path,
+                mode="100644",
+                type="blob",
+                sha=changed_file.sha,
+            )
+        )
+
+    # Create a new commit pointing to that tree
+    if write:
+        tree = repo.create_git_tree(tree_elements, base_tree=parent_tree)
+        commit = repo.create_git_commit(message, tree, [parent_commit])
+        repo.get_git_ref(f"heads/{new_branch_name}").edit(sha=commit.sha)
+
+    # Check if there's an existing pull request
+    found_pr = None
+    open_pulls = repo.get_pulls(state="open", base="master")
+    for pr in open_pulls:
+        if pr.head.ref == new_branch_name:
+            found_pr = pr
+            logger.info(f"        Pull request already exists: {pr.html_url}")
+
+    if found_pr:
+        pull_request_number = found_pr.number
+        if input_title and input_body:
+            logger.info("          Updating title and body")
+            if write:
+                found_pr.edit(title=input_title, body=input_body)
+        elif input_title:
+            logger.info("          Updating title")
+            if write:
+                found_pr.edit(title=input_title)
+        elif input_body:
+            logger.info("          Updating body")
+            if write:
+                found_pr.edit(body=input_body)
+    else:
+        # Create a pull request if it's a new branch
+        if not write:
+            pull_request_number = 0
+        else:
+            pull_request_title = input_title or f"{connector_full_name}: {message}"
+            pull_request_body = input_body or ""
+            pull_request = repo.create_pull(
+                title=pull_request_title,
+                body=pull_request_body,
+                base=BASE_BRANCH,
+                head=new_branch_name,
+            )
+
+            # TODO: could pass in additional labels
+            label = repo.get_label("autopull")
+            pull_request.add_to_labels(label)
+            logger.info(f"        Created pull request: {pull_request.html_url}")
+            pull_request_number = pull_request.number
+
+    return pull_request_number
+
+
+async def run_connector_pull_request(
+    context: ConnectorContext,
+    semaphore: "Semaphore",
+    message: str,
+    branch_id: str,
+    title: str | None,
+    body: str | None,
+    changelog: bool,
+    bump: str | None,
+    dry_run: bool,
+) -> Report:
+    restore_original_state = RestorePullRequestState(context)
+
+    context.targeted_platforms = [LOCAL_BUILD_PLATFORM]
+
+    connector_version: str | None = context.connector.version
+
+    steps_to_run: STEP_TREE = []
+
+    steps_to_run.append(
```

Comment:
Why are we first creating the PR?
A logical flow IMO would be to first bump version and modify changelog then push and open the PR.
But I might be missing something.

Reply:
The changelog needs the PR number that gets created.